keras/src/utils/saved_model_utils.py

```diff
@@ -0,0 +1,73 @@
+import json
+import zipfile
+
+import h5py
+
+from keras.saving import deserialize_keras_object
+from keras.src.saving.saving_lib import H5IOStore
+
+_CONFIG_FILENAME = "config.json"
+_METADATA_FILENAME = "metadata.json"
+_VARS_FNAME = "model.weights"
+
+
+class KerasFileEditor:
+    def __init__(self, filepath, reference_model=None, custom_objects=None):
+        self.filepath = filepath
+        self.custom_objects = custom_objects
+        self.metadata = None
+        self.reference_model = None
+        self.config = None
+
+        if filepath.endswith(".keras"):
+            self._init_for_keras_format(
+                custom_objects, filepath, reference_model
+            )
+            weights_store = H5IOStore(
+                _VARS_FNAME + ".h5",
+                archive=zipfile.ZipFile(filepath, "r"),
+                mode="r",
+            )
+        elif filepath.endswith(".weights.h5"):
+            weights_store = H5IOStore(filepath, mode="r")
+        else:
+            raise ValueError(
+                "Invalid filename: "
+                "expected a `.keras` `.weights.h5` extension. "
+                f"Received: filepath={filepath}"
+            )
+
+        def _extract_values(data):
+            result = {}
+            for key in data.keys():
+                value = data[key]
+                if isinstance(value, h5py.Group) and len(value) == 0:
+                    continue
+                if hasattr(value, "keys"):
+                    result[key] = _extract_values(value)
+                else:
+                    result[key] = value
+            return result
+
+        self.nested_dict = _extract_values(weights_store.h5_file)
+
+    def _init_for_keras_format(self, custom_objects, filepath, reference_model):
+        with zipfile.ZipFile(filepath, "r") as zf:
+            with zf.open(_CONFIG_FILENAME, "r") as f:
+                self.config = json.loads(f.read())
+            if reference_model is None:
+                self.reference_model = deserialize_keras_object(
+                    self.config, custom_objects=custom_objects
+                )
+
+            with zf.open(_METADATA_FILENAME, "r") as f:
+                self.metadata = json.loads(f.read())
+
+    def list_layer_paths(self):
```

Comment:
I implemented this function to display only the layers that contain non-empty weight values. However, I believe it might also be beneficial to show all layer paths, regardless of whether they have weights or not. I would appreciate your feedback on this approach.